tubemq-client-twins/tubemq-client-go/remote/remote.go

```diff
@@ -449,14 +445,20 @@ func (r *RmtDataCache) ReleasePartition(checkDelay bool, filterConsume bool, con
 }
 
 func (r *RmtDataCache) removeFromIndexPartitions(partitionKey string) {
-	pos := 0
+	pos := -1
 	for i, p := range r.indexPartitions {
 		if p == partitionKey {
 			pos = i
 			break
 		}
 	}
-	r.indexPartitions = append(r.indexPartitions[:pos], r.indexPartitions[pos+1:]...)
+	if len(r.indexPartitions) == 1 && pos == 0 {
+		r.indexPartitions = []string{}
+		return
+	}
+	if pos >= 0 {
+		r.indexPartitions = append(r.indexPartitions[:pos], r.indexPartitions[pos+1:]...)
```

Comment:
Need check pos is Last.
if len(r.indexPartitions) == 1 && pos == 0 {
   r.indexPartitions = nil  // Because always use append to add item.
}
if pos >= 0 {
....
}